### ISSUE TITLE
The game tries to load an unavailable ruleset if you select a beatmap of it

### ISSUE THREAD

**Issue by Gabixel** [ppy/osu#13999]
**Describe the bug:** While I was scrolling through the song list inside the Now Playing overlay, I noticed this empty beatmap:

![empty](https://user-images.githubusercontent.com/43073074/126869215-be2ec945-b31c-4fa3-be33-b56e6bdb1d7d.png)

---

```
2021-07-24 12:32:34 [verbose]: Focus contention triggered by NowPlayingOverlay.
2021-07-24 12:32:43 [verbose]: working beatmap updated to -  (Gabixel)
2021-07-24 12:32:43 [error]: An unhandled error has occurred.
2021-07-24 12:32:43 [error]: System.Collections.Generic.KeyNotFoundException: The given key 'karaoke!' was not present in the dictionary.
```

The runtime log says that it was made by me in the "karaoke!" ruleset (which I downloaded a while ago to try out unofficial rulesets).

I'm not 100% sure, but this empty beatmap was probably generated after I accidentally opened the editor with that karaoke ruleset as the active one.

In case you need it, here's the [`client.db`](https://drive.google.com/file/d/18G4VHL1Glr8Y0IZ832r5MMyMv5GWVIzY/view?usp=sharing) (is this the right file?).

It seems that the game tries to load this karaoke ruleset (even if it's not present at the moment):
```
2021-07-24 12:32:46 [verbose]: decoupled ruleset transferred ("osu!" -> "karaoke!")
```

You can see, in the following video, that the star rating disappears from each map, and the game throws various exceptions.
If you change the ruleset everything comes back to normal. But if you try to re-enter the carousel with the unavailable ruleset, you will only see a blank screen (until you change the ruleset again).

**Screenshots or videos showing encountered issue:** 

https://user-images.githubusercontent.com/43073074/126869086-ca2550b2-c1e7-4ada-be34-1d840e1ab832.mp4

**osu!lazer version:** `2021.724.1-lazer`

**Logs:** [runtime.log](https://github.com/ppy/osu/files/6872437/runtime.log)

**Comment by peppy** [comment]
This seems pretty critical. We shouldn't be allowing the global ruleset to be set to a ruleset which is unavailable in the first place. Individual components are not designed to handle this scenario.

**Comment by Gabixel** [comment]
This specific scenario still appears to be reproducible in the latest release (2021.809.0).

After testing how I got the empty beatmap, I can confirm that I generated it after entering the editor with an unsupported ruleset (as mentioned in the OP).

These are the steps to reproduce the issue:
1. Add any third-party ruleset not supported by the editor (I tested with [_tau_](https://github.com/naoei/tau) and [_karaoke_](https://github.com/karaoke-dev/karaoke)).
2. Open the editor.
3. The editor will give an error. After this error, the empty beatmap will be generated.
4. Return to the menu and close the game.
5. Remove the ruleset with which you opened the editor and generated the empty beatmap.
6. Restart the game.
7. Enter the carousel (solo).
8. Select the empty beatmap from the `NowPlayingOverlay`'s playlist.
